packages/ui-components/src/components/common/progress-bar.ts

```diff
@@ -0,0 +1,76 @@
+import { BaseElement } from '../../base';
+import { html, nothing } from 'lit';
+import { property } from 'lit/decorators.js';
+
+export class ProgressBar extends BaseElement {
+  @property({ type: Number })
+  currentStep = 0;
+  @property({ type: Number })
+  totalSteps = 1;
+  @property({ type: Boolean })
+  hideProgressText = false;
+  @property({ type: Boolean })
+  hideProgressBar = false;
+
+  tresholdLow = 60;
+  tresholdHight = 80;
+
+  backgroundColors: { [index: number]: string } = {
+    0: 'bg-red-600',
+    1: 'bg-yellow-400',
+    2: 'bg-green-600',
+  };
+  textColors: { [index: number]: string } = {
+    0: 'text-red-600',
+    1: 'text-yellow-400',
+    2: 'text-green-600',
+  };
+
+  getColor(): number {
+    if (this.currentStep / this.totalSteps <= this.tresholdLow / 100) {
+      return 0;
+    } else if (
+      this.currentStep / this.totalSteps > this.tresholdLow / 100 &&
+      this.currentStep / this.totalSteps <= this.tresholdHight / 100
+    ) {
+      return 1;
+    } else {
+      return 2;
+    }
+  }
+
+  render() {
+    const progressText = html`
+      <span
+        class="font-bold ${this.textColors[this.getColor()]} 
+          ${this.hideProgressBar ? '' : 'pr-3'}"
+        ?hidden="${this.hideProgressText}"
+      >
+        ${Math.round((this.currentStep / this.totalSteps) * 1000) / 10}%
+      </span>
+    `;
+    const progressBar = html`
+      <div
+        class="bg-neutral-600 rounded-full h-4 w-full"
+        ?hidden="${this.hideProgressBar}"
```

Comment:
It means that it will add the attribute based on the outcome of the boolean.

If it is true then `hidden` will be added to the DOM, if not it won't be. Not sure where to find this since I asked ChatGPT 😆

Reply:
https://lit.dev/docs/templates/expressions/#boolean-attribute-expressions

Reply:
cool! learned something new 😯